app/controllers/find/courses_controller.rb

```diff
@@ -12,6 +12,8 @@ class CoursesController < ApplicationController
     before_action :set_search_params, only: :show
 
     def show
+      @coordinates = Geolocation::CoordinatesQuery.new(params[:location]).call if params[:location]
```

Comment:
Is it not better move the call to the nearest school also in the controller rather than the component?

Reply:
Wdyt?

Reply:
Then we can pass to the component the hash with coordinates and the minimum distance. The component won't touch in the DB